File: server/models/Product.js
const mongoose = require('mongoose');

const productStatus = require('./productStatus.js');

const ProductSchema = new mongoose.Schema(
  {
    //id will be set by default

    title: {
      type: mongoose.Schema.Types.String,
      required: true,
    },
    description: {
      type: mongoose.Schema.Types.String,
      required: true,
    },
    vendor: {
      type: mongoose.Schema.Types.String,
      required: true,
    },
    createdAt: {
      type: mongoose.Schema.Types.Date,
      default: Date.now,
    },
    updatedAt: {
      type: mongoose.Schema.Types.Date,
      default: Date.now,
    },
    publishedAt: {
      type: mongoose.Schema.Types.Date,
      default: Date.now,
    },
    status: {
      type: mongoose.Schema.Types.String,
      enum: [productStatus.available, productStatus.outOfStock], //TODO clean this hard coded value
      // required: true,
    },
    tags: [
      {
        type: mongoose.Schema.Types.String,
      },
    ],
    weight: {
      type: mongoose.Schema.Types.String,
      // required: true,
    },
    weight_unit: {
      type: mongoose.Schema.Types.String,
      // required: true,
    },
    price: {
      type: mongoose.Schema.Types.Number,
      required: true,
    },
    amount: {
      type: mongoose.Schema.Types.String,
      // required: true,
    },
    category: {
      type: mongoose.Schema.Types.String,
    },
    images: [{ type: mongoose.Schema.Types.String }],
  },
  {
    writeConcern: {
      w: 'majority',
      j: true,
      wtimeout: 1000,
    },
  }
);

const Product = mongoose.model('Product', ProductSchema);

module.exports = Product;

File: application/RaneenXamarinProject/RaneenXamarinProject/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using System.Text;

namespace RaneenXamarinProject.Models
{
    public class Product: INotifyPropertyChanged
    {
        public string _id { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public string vendor { get; set; }
        public DateTime created_at { get; set; }
        public DateTime updated_at { get; set; }
        public DateTime published_at { get; set; }
        public string status { get; set; }
        public string category { get; set; }
        public List<string> tags { get; set; }
        public string admin_id { get; set; }
        public float weight { get; set; }
        public float weight_unit { get; set; }
        public float price { get; set; }
        public int amount { get; set; }
        public List<string> images { get; set; }

        private bool isFavourite;

        public bool IsFavourite
        {
            get
            {
                return isFavourite;
            }
            set
            {
                isFavourite = value;
                this.NotifyPropertyChanged("IsFavourite");
            }
        }

        public double DiscountPrice 
        { 
            get
            {
                return this.price - (this.price * (this.DiscountPercent/100));
            }
        }
        public double DiscountPercent { get; } = 12;

        public double OverallRating { get; } = 4;

        public event PropertyChangedEventHandler PropertyChanged;

        private void NotifyPropertyChanged([CallerMemberName] string propertyName = null)
        {
            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        private bool SetProperty<T>(ref T storage, T value, [CallerMemberName] string propertyName = null)
        {
            if (EqualityComparer<T>.Default.Equals(storage, value))
            {
                return false;
            }

            storage = value;
            this.NotifyPropertyChanged(propertyName);

            return true;
        }


    }

    public class CartItem
    {
        public string id { get; set; }

        public int amount { get; set; } = 1;

        public Product item { get; set; }

    }
}
